common/src/com/thoughtworks/go/domain/materials/git/GitCommand.java

```diff
@@ -59,7 +59,11 @@ public int cloneFrom(ProcessOutputStreamConsumer outputStreamConsumer, String ur
     // Clone repository from url with specified depth.
     // Special depth 2147483647 (Integer.MAX_VALUE) are treated as full clone
     public int cloneFrom(ProcessOutputStreamConsumer outputStreamConsumer, String url, Integer depth) {
-        CommandLine gitClone = git().withArg("clone").withArg(String.format("--branch=%s", branch));
+        CommandLine gitClone = git()
```

Comment:
@ketan 
On agent we always do a hard reset to a specific revision, so the initial working copy checkout is not needed and just wasting time.